classes/mapworker.php

```diff
@@ -294,4 +298,33 @@ public function process_map_objects(\cm_info $cm) : void {
     public function get_svgcode(): string {
         return $this->svgcode;
     }
+
+    /**
+     * Checks whether a given course module is completed (either by the user or at least one
+     * of the users of the group, if groupmode is set for the activity).
+     *
+     * @param \cm_info $cm course module to check
+     */
+    public function is_completed(\cm_info $cm): bool {
```

Comment:
Pretty similar to the function with same name in `custom_completion.php`. Any chance merging these?

Reply:
I'd rather move that question to the refactoring of some parts.